src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
I haven't looked yet, and maybe @briansull should chime in, but in general, you need to be *very* careful with locking registers, especially if an unbounded amount of code generation can happen while they are locked.